node-wasm/src/client.rs

```diff
@@ -409,3 +409,116 @@ impl WasmNodeConfig {
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::time::Duration;
+
+    use super::*;
+    use crate::worker::NodeWorker;
+    use celestia_rpc::{prelude::*, Client};
+    use celestia_types::ExtendedHeader;
+    use libp2p::{multiaddr::Protocol, Multiaddr};
+    use rexie::Rexie;
+    use serde_wasm_bindgen::from_value;
+    use wasm_bindgen_futures::spawn_local;
+    use wasm_bindgen_test::wasm_bindgen_test;
+    use web_sys::MessageChannel;
+
+    // uses bridge-0, which has skip-auth enabled
+    const WS_URL: &str = "ws://127.0.0.1:36658";
+
+    #[wasm_bindgen_test]
+    async fn request_network_head_header() {
+        remove_database().await.expect("failed to clear db");
+        let rpc_client = Client::new(WS_URL).await.unwrap();
+        let bridge_ma = fetch_bridge_webtransport_multiaddr(&rpc_client).await;
+
+        let client = spawn_connected_node(vec![bridge_ma.to_string()]).await;
+
+        let info = client.network_info().await.unwrap();
+        assert_eq!(info.num_peers, 1);
+
+        let bridge_head_header = rpc_client.header_network_head().await.unwrap();
+        let head_header: ExtendedHeader =
+            from_value(client.request_head_header().await.unwrap()).unwrap();
+        assert_eq!(head_header, bridge_head_header)
+    }
+
+    #[wasm_bindgen_test]
+    async fn discover_network_peers() {
+        remove_database().await.expect("failed to clear db");
+        let rpc_client = Client::new(WS_URL).await.unwrap();
+        let bridge_ma = fetch_bridge_webtransport_multiaddr(&rpc_client).await;
+
+        let client = spawn_connected_node(vec![bridge_ma.to_string()]).await;
+
+        let info = client.network_info().await.unwrap();
+        assert_eq!(info.num_peers, 1);
+
+        gloo_timers::future::sleep(Duration::from_secs(2)).await;
```

Comment:
```suggestion
        gloo_timers::future::sleep(Duration::from_millis(300)).await;
```
I think we can reduce this now that we solved other issues

Reply:
+ nit, maybe import the sleep